File: Movs/Movs/Models/Movie.swift
//
//  Movie.swift
//  Movs
//
//  Created by Eduardo Pereira on 26/08/19.
//  Copyright © 2019 Eduardo Pereira. All rights reserved.
//

import Foundation

struct APIRequest<T:Decodable>:Decodable{
    let results:[T]
}

struct Movie:Decodable {
    //parameters needed
    let id:Int
    let title:String?
    let popularity:Float?
    let poster_path:String?
    let genres:[Genre]?
    let backdrop_path:String?
    let overview:String?
    let release_date:String?
    let runtime:Int?
    let videos:APIRequest<Video>?
}

File: MoviesMobileChallenge/app/src/main/java/br/com/suelen/mobilechallenge/data/model/Movie.kt
package br.com.suelen.mobilechallenge.data.model

data class Movie(
    var id : Int,
    var title : String?,
    var release_date : String?,
    var runtime : Int,
    var genres : List<Genre>?,
    var overview : String?,
    var poster_path : String?,
    var backdrop_path : String?)